Mathlib/Combinatorics/SimpleGraph/Diam.lean

```diff
@@ -103,11 +176,19 @@ lemma preconnected_of_ediam_ne_top (h : G.ediam ≠ ⊤) : G.Preconnected :=
 lemma connected_of_ediam_ne_top [Nonempty α] (h : G.ediam ≠ ⊤) : G.Connected :=
   G.connected_iff.mpr ⟨preconnected_of_ediam_ne_top h, ‹_›⟩
 
+lemma exists_eccent_eq_ediam_of_ne_top [Nonempty α] (h : G.ediam ≠ ⊤) :
+    ∃ u, G.eccent u = G.ediam :=
+  ENat.exists_eq_iSup_of_lt_top h.lt_top
+
+-- Note: Neither `Finite α` nor `G.ediam ≠ ⊤` implies the other.
+lemma exists_eccent_eq_ediam_of_finite [Nonempty α] [Finite α] :
+    ∃ u, G.eccent u = G.ediam :=
+  exists_eq_ciSup_of_finite
+
 lemma exists_edist_eq_ediam_of_ne_top [Nonempty α] (h : G.ediam ≠ ⊤) :
     ∃ u v, G.edist u v = G.ediam :=
   ENat.exists_eq_iSup₂_of_lt_top h.lt_top
 
--- Note: Neither `Finite α` nor `G.ediam ≠ ⊤` implies the other.
```

Comment:
I think we can keep this one; it's a duplicate of the earlier one but I think that's fine